WeightedEuclidean.lua

```diff
@@ -25,55 +17,63 @@ function WeightedEuclidean:reset(stdv)
    if stdv then
       stdv = stdv * math.sqrt(3)
    else
-      stdv = 1./math.sqrt(self.templates:size(1))
-   end
-   if nn.oldSeed then
-      for i=1,self.templates:size(2) do
-         self.templates:select(2, i):apply(function()
-            return torch.uniform(-stdv, stdv)
-         end)
-      end
-   else
-      self.templates:uniform(-stdv, stdv)
+      stdv = 1./math.sqrt(self.weight:size(1))
    end
+   self.weight:uniform(-stdv, stdv)
    self.diagCov:fill(1)
 end
 
+
 function WeightedEuclidean:updateOutput(input)
    -- lazy-initialize 
-   self._temp = self._temp or self.output.new()
-   self._ones = self._ones or self.output.new{1}
    self._diagCov = self._diagCov or self.output.new()
+   
+   self._input = self._input or input.new()
+   self._weight = self._weight or self.weight.new()
+   self._expand = self._expand or self.output.new()
+   self._expand2 = self._expand or self.output.new()
+   self._expand3 = self._expand3 or self.output.new()
    self._repeat = self._repeat or self.output.new()
-   self._sum = self._sum or self.output.new()
-   self._temp:resizeAs(input)
+   self._repeat2 = self._repeat2 or self.output.new()
+   self._repeat3 = self._repeat3 or self.output.new()
+   
+   local inputSize, outputSize = self.weight:size(1), self.weight:size(2)
+   
+   -- y_j = || c_j * (w_j - x) ||
    if input:dim() == 1 then
-      self.output:resize(self.templates:size(2))
-      for outIdx = 1,self.templates:size(2) do
-         self._temp:copy(input):add(-1,self.templates:select(2,outIdx))
-         self._temp:cmul(self._temp)
-         local diagCov = self.diagCov:select(2,outIdx)
-         self._temp:cmul(diagCov):cmul(diagCov)
-         self.output[outIdx] = math.sqrt(self._temp:sum())
-      end
+      self._input:reshape(input, inputSize, 1)
```

Comment:
same comments as from the Euclidean PR. we should try to avoid reshape (and resize when you want to retain the memory, resize can be destructive as needed, does not guarantee that the memory stays the same, unless the underlying storage is not being resized. in this case, the storage is not being changed, but a good practice to use :view)